src/sentry/flags/providers.py

```diff
@@ -390,7 +384,39 @@ def handle_flag_pole_event_internal(items: list[FlagAuditLogItem], organization_
 """Helpers."""
 
 
-class SecretValidator:
+class AuthTokenValidator:
+    """Abstract payload validator.
+
+    Similar to the SecretValidator class below, except we do not need
+    to validate the authorization string.
+    """
+
+    def __init__(
+        self,
+        organization_id: int,
+        provider: str,
+        request_body: bytes,
+        signature: str | None,
+        secret_finder: Callable[[int, str], Iterator[str]] | None = None,
+    ) -> None:
+        self.organization_id = organization_id
+        self.provider = provider
+        self.request_body = request_body
```

Comment:
We don't need the request_body for this class.

Reply:
removed